numpy/core/tests/test_multiarray.py

```diff
@@ -3139,6 +3139,27 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kw):
         assert_equal(A[0], 30)
         assert_(isinstance(A, OutClass))
 
+    def test_pow_override_with_errors(self):
+        # regression test for gh-9112
+        class PowerOnly(np.ndarray):
+            def __array_ufunc__(self, ufunc, method, *inputs, **kw):
+                if ufunc is not np.power:
+                    raise NotImplementedError
```

Comment:
How would we detect that? We could need to actually call the `np.power` ufunc, which the code currently does not do.